ydb/docs/en/core/reference/ydb-sdk/_includes/auth.md

```diff
@@ -94,10 +96,41 @@ The following algorithm that is the same for all SDKs applies:
 2. Otherwise, if the value of the `YDB_ANONYMOUS_CREDENTIALS` environment variable is set to 1, the anonymous authentication mode is used.
 3. Otherwise, if the value of the `YDB_METADATA_CREDENTIALS` environment variable is set to 1, the **Metadata** authentication mode is used.
 4. Otherwise, if the value of the `YDB_ACCESS_TOKEN_CREDENTIALS` environment variable is set, the **Access token** authentication mode is used, where the this variable value is passed.
-5. Otherwise, the **Metadata** authentication mode is used.
+5. Otherwise, if the value of the `YDB_OAUTH2_KEY_FILE` environment variable is set, the **OAuth 2.0 token exchange** authentication mode is used and the parameters are taken from the [json-file](#oauth2-key-file-format) whose name is specified in this variable.
+6. Otherwise, the **Metadata** authentication mode is used.
 
 If the last step of the algorithm is selecting the **Metadata** mode, you can deploy a working application on VMs and in {{ yandex-cloud }} Cloud Functions without setting any environment variables.
 
+## File format that is used for OAuth 2.0 token exchange authentication mode parameters {#oauth2-key-file-format}
```

Comment:
This section needs some examples. Otherwise, there's no chance to write such a file from scratch.